### ISSUE TITLE
Can't hide OSC

### ISSUE THREAD

**Issue by Issue-maker-9000** [ripose-jp/Memento#113]
Hi first of all absolutely love the player, really helps with watching way more anime and youtube.

I just have a minor annoyance I just can't figure out in mpv I run a little [script](https://github.com/linguisticmind/mpv-scripts/blob/master/osc-show-hide-hotkey/osc-show-hide-hotkey.lua) which I slightly modified so I can use '/' to switch between OSC "hidden" and "always" on the fly and by default the OSC is hidden. Problem is I can't get it to work with memento at all, hotkeys like cycle visibility with DEL do work but the script somehow doesn't without any errors in the console. 

I might just overlooking something really obvious..... Help would be really appreciated

**Comment by ripose-jp** [comment]
The OSC key in Memento is not rebindable or scriptable.

This is because OSCs in mpv aren't actually built into the player like other options, but are actually lua scripts. This isn't a problem in and of itself, but unless `osc=yes` is set, these script events are never triggered. Setting `osc=yes` creates another problem where there are now two OSCs, Memento's and mpv's visible. Due to this, I made the choice to allow for cycling through OSC visibilities using the delete key without interacting with mpv beyond printing a message on the screen.

Basically the OSC is implemented in C++ and mpv scripts cannot modify it.

**Comment by ripose-jp** [comment]
Closing this due to inactivity.

**Comment by ripose-jp** [comment]
The original plan when I started development was to keep mpv's OSC, but that was harder than just making my own for a few reasons.

It is possible to replace Memento's OSC with mpv's by hiding Memento's OSC and then unhiding mpv's, but I would have to change the code in Memento to pass mouse movements to mpv. The only reason I don't do this now is because I don't want to take up CPU time on the UI thread just passing `mouse <x> <y>` commands to mpv since that would probably get decently expensive with creating strings and having mpv parsing them. That said, I haven't done any performance testing so that's just speculation.

If I could replace the baked in mpv OSC with a dummy OSC that just passes script events to Memento, I could make the Memento OSC scriptable, but I haven't really looked into it.

Really the reason I don't think this is such a big deal is because the solution is just press a key twice instead of once.

**Comment by Issue-maker-9000** [comment]
Tbf my thinking was that it would also allow plugins like [MPV thumbnail](https://github.com/marzzzello/mpv_thumbnail_script) to work for example, and like every other plugin and config setting in existence. (Tbf wasted way more time why plugins weren't working but I also use way too many plugins in MPV)

Although it indeed sounds like more work than it is worth besides probably people with weak computers who would actually suffer from degraded performance, in the end might be better to keep it this way at least the Memento OSC looks nice. ;)

Main annoyance was indeed the OSC you can't hide by default but it is what it is, won't change my mind from using this memento in anyway. xD

**Comment by ripose-jp** [comment]
I tried to implement passing mouse events to mpv just now and remembered why I didn't do it in the first place. Even though I was passing mouse clicks to the player, I wasn't able to actually press any buttons on mpv's OSC. I also found that dragging can't really be emulated cleanly either. libmpv's API wasn't designed for the default OSC to be used unfortunately. The only libmpv player that I know of that gets around these constraints is mpv.net, but I'm sure they handle a ton of edge cases to get it to work. Honestly I don't really think it's worth it to try and get custom OSCs to work.

Other people are certainly welcome to make a PR.

>you can't hide by default

The only reason I don't have an option to do this is because I'm afraid some user is going to set it and then not know how to open the OSC again to unset it. Hiding it is just two keypresses, so I don't think it's a big tradeoff.
